### PR TITLE
feat: regexp feature

### PR DESCRIPTION
**Pull request by load1n9**

### CODE REVIEW

File: nova_vm/Cargo.toml [hunk 1]
```diff
@@ -24,16 +24,26 @@ unicode-normalization = { workspace = true }
...
 atomics = ["array-buffer", "shared-array-buffer"]
 date = []
 interleaved-gc = []
 json = ["sonic-rs"]
 math = []
 shared-array-buffer = []
-typescript = []
 weak-refs = []
+regexp = []
+typescript = []
```
Comment (aapoalas) [inline nova_vm/Cargo.toml:46]: **question:** Is there a reason to keep typescript at the bottom?

I sorted these into alphabetical order (except for the default features) the other week.

Comment (load1n9) [reply nova_vm/Cargo.toml:46]: lmao no reason didnt realize you did that, was a slight nitpick mentally since its the only one not used by default

Comment (load1n9) [reply nova_vm/Cargo.toml:46]: would you prefer i change it back? might make more sense

Comment (aapoalas) [reply nova_vm/Cargo.toml:46]: Regexp at least should be alphabetical. TS can be last as a "non-default feature"

File: nova_vm/Cargo.toml [hunk 2]
```diff
@@ -24,16 +24,26 @@ unicode-normalization = { workspace = true }
...
 
 [features]
-default = ["math", "json", "date", "array-buffer", "shared-array-buffer", "weak-refs", "atomics"]
+default = [
+    "math",
+    "json",
+    "date",
+    "array-buffer",
+    "shared-array-buffer",
+    "weak-refs",
```
Comment (aapoalas) [inline nova_vm/Cargo.toml:33]: **nitpick:** Could sort these alphabetically as well.

**Review by aapoalas — DISMISSED** [review]
LGTM, though there's a merge conflict in heap.rs

**Review by aapoalas — APPROVED** [review]

**Pull request merged**